src/fetch.js

```diff
@@ -260,34 +262,52 @@ const getSyncData = async (
       ? response.sync_token
       : aggregatedResponse.sync_token;
   }
-  if (response.pagination_token) {
-    return getSyncData(
-      url,
-      config,
-      (query = { pagination_token: response.pagination_token }),
-      responseKey,
-      aggregatedResponse
-    );
-  }
+    if (response.pagination_token) {
+      try {
+        return await getSyncData(
+          url,
+          config,
+          { pagination_token: response.pagination_token },
+          responseKey,
+          aggregatedResponse,
+          0 // Reset retries for each call
+        );
+      } catch (error) {
+        if (retries < config.httpRetries) {
+          const timeToWait = 2 ** retries * 100;
+          console.log(`Retry attempt ${retries + 1} after pagination token error. Waiting for ${timeToWait} ms...`);
+          await waitFor(timeToWait);
+          return await getSyncData(
+            url,
+            config,
+            { pagination_token: response.pagination_token },
+            responseKey,
+            aggregatedResponse,
+            retries + 1
+          );
+        } else {
+          throw new Error(`Failed to fetch sync data after ${config.httpRetries} retry attempts due to invalid pagination token.`);
+        }
+      }
+    }
 
-  if (response.sync_token) {
-    /**
-     * To make final sync call and concatenate the result if found any during on fetch request.
-     */
+    if (response.sync_token) {
     const aggregatedSyncToken = syncToken.filter(item => item !== undefined);
     for (const token of aggregatedSyncToken) {
       const syncResponse = await fetchCsData(
         url,
         config,
-        (query = { sync_token: token })
-      );
-      aggregatedResponse.data = aggregatedResponse.data?.concat(
-        ...syncResponse.items
+        { sync_token: token }
       );
+      aggregatedResponse.data = aggregatedResponse.data.concat(syncResponse.items);
```

Comment:
this is where we aggregate all the data make sure no entries are missing while quering via graphql.
I suspect removing spread operator would result in data loss.